src/compose/service.ts

```diff
@@ -1124,26 +1147,50 @@ export class Service {
 	): ServiceConfig['volumes'] {
 		let volumes: ServiceConfig['volumes'] = [];
 
-		_.each(composeVolumes, (volume) => {
-			const isBind = _.includes(volume, ':');
-			if (isBind) {
-				const [bindSource, bindDest, mode] = volume.split(':');
-				if (!path.isAbsolute(bindSource)) {
-					// namespace our volumes by appId
-					let volumeDef = `${appId}_${bindSource.trim()}:${bindDest.trim()}`;
+		// namespace our volumes by appId
+		const namespaceVolume = (volumeSource: string) =>
+			`${appId}_${volumeSource.trim()}`;
+
+		for (const volume of composeVolumes || []) {
+			const isAnonymousOrTmpfs =
+				(LongSyntaxVolume.is(volume) && volume.type === 'tmpfs') ||
+				(typeof volume === 'string' && !volume.includes(':'));
+
+			let isBindMount = false;
+
+			if (isAnonymousOrTmpfs) {
+				// Push anonymous volume. Anonymous volumes can only be
+				// set by using the `VOLUME` instruction inside a Dockerfile.
+				// Tmpfs volumes do not require namespacing by appId.
+				volumes.push(volume);
+			} else if (LongSyntaxVolume.is(volume)) {
+				// Handle long syntax volumes of type 'tmpfs' and 'volume'
+				isBindMount = path.isAbsolute(volume.source!);
```

Comment:
If `volume.type` is `tmpfs`, then `volume.source` will be undefined, meaning the call to `path.isAbsolute` will fail, right??

```
path.isAbsolute(undefined)
Uncaught:
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at validateString (internal/validators.js:120:11)
    at Object.isAbsolute (path.js:1029:5)
    at REPL10:1:6
    at Script.runInThisContext (vm.js:120:18)
    at REPLServer.defaultEval (repl.js:442:29)
    at bound (domain.js:427:14)
    at REPLServer.runBound [as eval] (domain.js:440:12)
    at REPLServer.onLine (repl.js:777:10)
    at REPLServer.emit (events.js:326:22)
    at REPLServer.EventEmitter.emit (domain.js:483:12) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

Reply:
I clarified the comments in the latest push, but the path.isAbsolute calls should only be for long syntax volumes of type `bind` and `volume`, since `tmpfs` long syntax volumes are handled in the first `if` at the same code branch as anonymous volumes. 

I did notice that compose's behavior allows for long syntax volumes without a `source` defined (I suppose they behave as anonymous volumes), so I've updated the `if`s to reflect this.